ext/cron_uploader/main.php

```diff
@@ -273,6 +273,11 @@ public function process_upload(int $upload_count = 0): bool
                 $this->move_uploaded($img[0], $img[1], false);
             } catch (Exception $e) {
                 $this->move_uploaded($img[0], $img[1], true);
+                if (strpos($e->getMessage(), 'SQLSTATE') !== false) {
```

Comment:
I wonder what SQL errors can happen? Is there some way to avoid hitting them in the first place?

Reply:
Basically just bugs. The only error I encountered was a primary key violation caused by the item merge code sending duplicate tags to be written. The array unique change I submitted in the other pull request resolved most of those, but I found some other ways to trigger it that I'm going to make a ticket to discuss in.